spec/index.bs

```diff
@@ -11985,6 +12002,132 @@ It must only be included by interfaces which also include those mixins.
+            let drawIndirectParameters = new Uint32Array(8);
+            // first draw
+            drawIndirectParameters[0] = vertexCount0;
+            drawIndirectParameters[1] = instanceCount0;
+            drawIndirectParameters[2] = firstVertex0;
+            drawIndirectParameters[3] = firstInstance0;
+            // second draw
+            drawIndirectParameters[4] = vertexCount1;
+            drawIndirectParameters[5] = instanceCount1;
+            drawIndirectParameters[6] = firstVertex1;
+            drawIndirectParameters[7] = firstInstance1;
+        </pre>
+
+        The <dfn dfn for=>indirect multiDraw count</dfn> encoded in the |drawCountBuffer| must be a **32-bit unsigned
+        integer** giving the number of draws to issue from the |indirectBuffer|.
+
+        <div algorithm="GPURenderCommandsMixin.multiDrawIndirect">
+            **Called on:** {{GPURenderCommandsMixin}} this.
+
+            **Arguments:**
+            <pre class=argumentdef for="GPURenderCommandsMixin/multiDrawIndirect(indirectBuffer, indirectOffset, maxDrawCount, drawCountBuffer, drawCountOffset)">
+                |indirectBuffer|: Buffer containing the [=indirect multiDraw parameters=].
+                |indirectOffset|: Offset in bytes into |indirectBuffer| where the drawing data begins.
+                |maxDrawCount|: Maximum number of draws or exact number of draws if |drawCountBuffer| is `null`.
+                |drawCountBuffer|: Buffer containing the [=indirect multiDraw count=].
+                |drawCountOffset|: Offset in bytes into |drawCountBuffer| where the [=indirect multiDraw count=] is located.
+            </pre>
+
+            **Returns:** {{undefined}}
+
+            Issue the following steps on the [=Device timeline=] of |this|.{{GPUObjectBase/[[device]]}}:
+            <div class=device-timeline>
+                1. If any of the following conditions are unsatisfied, make |this| [=invalid=] and stop.
+                    <div class=validusage>
+                        - It is [$valid to draw$] with |this|.
+                        - |indirectBuffer| is [$valid to use with$] |this|.
+                        - |indirectBuffer|.{{GPUBuffer/[[usage]]}} contains {{GPUBufferUsage/INDIRECT}}.
+                        - |indirectOffset| + sizeof([=indirect multiDraw parameters=]) &le;
+                            |indirectBuffer|.{{GPUBuffer/[[size]]}}.
+                        - |indirectOffset| is a multiple of 4.
+                        - |maxDrawCount| &le; limits.{{supported limits/maxMultiDrawIndirectCount}}.
+                        - |drawCountBuffer| is [$valid to use with$] |this|.
+                        - |drawCountBuffer|.{{GPUBuffer/[[usage]]}} contains {{GPUBufferUsage/INDIRECT}}.
+                        - |drawCountOffset| + sizeof([=indirect multiDraw count=]) &le;
+                            |drawCountBuffer|.{{GPUBuffer/[[size]]}}.
+                        - |drawCountOffset| is a multiple of 4.
```

Comment:
Would it make sense to allow any value of `drawCountOffset` if `drawCountBuffer` is not set?

Reply:
I would probably just keep it, but we could also consider using overloads instead so that we can never have a `drawCountOffset` when there is no `drawCountBuffer`.

Reply:
@kainino0x FYI I'm asking because of https://github.com/gpuweb/cts/pull/3962/files#diff-faa22eb9a32ec5f94def60bb63172dfef5a8907fd5102d49a8203c762d8ee57cR214

Reply:
I don't know what we'll do here but two assorted notes:
- [setVertexBuffer](https://gpuweb.github.io/gpuweb/#dom-gpurendercommandsmixin-setvertexbuffer) still validates the buffer offset when buffer is null
- I like the overload solution best but we should think about what it will look like in languages without overloads (C webgpu.h). I think we currently only have one overloaded method in the API (`setBindGroup`) and that overload doesn't apply to C at all.
  - Two functions? Merge into one function where if drawCountBuffer is null, drawCountOffset {is ignored,must be 0,etc}